packages/camera/camera/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -367,276 +471,278 @@ private void createCaptureSessionWithSessionConfig(
+
+      // Update builder settings
+      updateBuilderSettings(stillBuilder);
+
+      // Orientation
+      final PlatformChannel.DeviceOrientation lockedOrientation =
+          ((SensorOrientationFeature) cameraFeatures.getSensorOrientation())
+              .getLockedCaptureOrientation();
+      stillBuilder.set(
           CaptureRequest.JPEG_ORIENTATION,
-          lockedCaptureOrientation == null
-              ? deviceOrientationListener.getMediaOrientation()
-              : deviceOrientationListener.getMediaOrientation(lockedCaptureOrientation));
-
-      switch (flashMode) {
-        case off:
-          captureBuilder.set(CaptureRequest.CONTROL_AE_MODE, CaptureRequest.CONTROL_AE_MODE_ON);
-          captureBuilder.set(CaptureRequest.FLASH_MODE, CaptureRequest.FLASH_MODE_OFF);
-          break;
-        case auto:
-          captureBuilder.set(
-              CaptureRequest.CONTROL_AE_MODE, CaptureRequest.CONTROL_AE_MODE_ON_AUTO_FLASH);
-          break;
-        case always:
-        default:
-          captureBuilder.set(
-              CaptureRequest.CONTROL_AE_MODE, CaptureRequest.CONTROL_AE_MODE_ON_ALWAYS_FLASH);
-          break;
-      }
-      cameraCaptureSession.stopRepeating();
-      cameraCaptureSession.capture(
-          captureBuilder.build(),
+          lockedOrientation == null
+              ? getDeviceOrientationManager().getPhotoOrientation()
+              : getDeviceOrientationManager().getPhotoOrientation(lockedOrientation));
+
+      CameraCaptureSession.CaptureCallback captureCallback =
           new CameraCaptureSession.CaptureCallback() {
+
+            @Override
+            public void onCaptureStarted(
+                @NonNull CameraCaptureSession session,
+                @NonNull CaptureRequest request,
+                long timestamp,
+                long frameNumber) {
+              Log.i(TAG, "onCaptureStarted");
+            }
+
+            @Override
+            public void onCaptureProgressed(
+                @NonNull CameraCaptureSession session,
+                @NonNull CaptureRequest request,
+                @NonNull CaptureResult partialResult) {
+              Log.i(TAG, "onCaptureProgressed");
+            }
+
             @Override
             public void onCaptureCompleted(
                 @NonNull CameraCaptureSession session,
                 @NonNull CaptureRequest request,
                 @NonNull TotalCaptureResult result) {
+              Log.i(TAG, "onCaptureCompleted");
               unlockAutoFocus();
             }
-          },
-          null);
+          };
+
+      captureSession.stopRepeating();
+      captureSession.abortCaptures();
+      Log.i(TAG, "sending capture request");
+      captureSession.capture(stillBuilder.build(), captureCallback, backgroundHandler);
     } catch (CameraAccessException e) {
-      pictureCaptureRequest.error("cameraAccess", e.getMessage(), null);
+      dartMessenger.error(flutterResult, "cameraAccess", e.getMessage(), null);
```

Comment:
please have simple logic in a try block.  rule of thumb: only have the code that throws `CameraAccessException`.